test/testrunner.py

```diff
@@ -130,11 +130,15 @@ def setup_eks_clusters(dlc_images):
         )
     long_name = frameworks_in_images[0]
     short_name = frameworks[long_name]
-    num_nodes = 2 if is_pr_context() else 3 if long_name != "pytorch" else 4
+    num_nodes = 1 if is_pr_context() else 3 if long_name != "pytorch" else 4
     cluster_name = f"dlc-{short_name}-cluster-" \
                    f"{os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION')}-{random.randint(1, 10000)}"
-    eks_utils.create_eks_cluster(cluster_name, "gpu", num_nodes, "p3.16xlarge", "pytest.pem")
-    eks_utils.eks_setup(long_name, cluster_name)
+    try:
+        eks_utils.create_eks_cluster(cluster_name, "gpu", num_nodes, "p3.16xlarge", "pytest.pem")
+        eks_utils.eks_setup(long_name, cluster_name)
+    except Exception as e:
+        eks_utils.delete_eks_cluster(cluster_name)
+        raise RuntimeError(e)
```

Comment:
This is a modification that comes from a change request on an older PR - https://github.com/aws/deep-learning-containers/pull/218#discussion_r432798693

Reply:
nit: just raise the existing exception instead of raising it as a runtime error